modules/library/src/main/scala/zio/elasticsearch/SubAggregation.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2022 LambdaWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.elasticsearch
+
+import zio.elasticsearch.ElasticAggregation.TermsAggregation
+import zio.elasticsearch.ElasticAggregationType.Terms
+
+object SubAggregation {
+
+  trait WithSubAggregation[EAT <: ElasticAggregationType] {
+    def withSubAgg(
+      aggregation: ElasticAggregation[EAT],
+      subAggregation: ElasticAggregation[_ <: SingleElasticAggregationType]
+    ): ElasticAggregation[EAT]
+  }
+
+  object WithSubAggregation {
+    implicit val termsWithSubAggregation: WithSubAggregation[Terms] =
+      (aggregation: ElasticAggregation[Terms], subAggregation: ElasticAggregation[_ <: SingleElasticAggregationType]) =>
+        aggregation match {
+          case a: TermsAggregation =>
```

Comment:
Are we going to add other aggregations here? Maybe we should use `termsAgg: TermsAggregation`.